pkg/controller/endpointslice/metrics/cache.go

```diff
@@ -128,34 +146,21 @@ type metricsUpdate struct {
 	desired, actual int
 }
 
-// desiredAndActualSlices returns a metricsUpdate with the desired and actual
-// number of EndpointSlices given the current values in the cache.
-// Must be called holding lock.
-func (c *Cache) desiredAndActualSlices() metricsUpdate {
-	mUpdate := metricsUpdate{}
-	for _, spCache := range c.cache {
-		for _, eInfo := range spCache.items {
-			mUpdate.actual += eInfo.Slices
-			mUpdate.desired += numDesiredSlices(eInfo.Endpoints, int(c.maxEndpointsPerSlice))
-		}
-	}
-	return mUpdate
-}
-
 // updateMetrics updates metrics with the values from this Cache.
 // Must be called holding lock.
-func (c *Cache) updateMetrics() {
-	mUpdate := c.desiredAndActualSlices()
+func (c *Cache) updateMetrics(mUpdate metricsUpdate) {
+	c.numSlicesActual = mUpdate.actual
+	c.numSlicesDesired = mUpdate.desired
```

Comment:
this does more things than updating metrics now.
Is it better to do the assignment just after we calculate the mUpdate? at the cost of duplicating code

Reply:
https://github.com/kubernetes/kubernetes/pull/107091/files#r771283831

Reply:
Yep, good call, that is cleaner + simpler.